### PR TITLE
ref(ts) Convert OrganizationStore to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 1]
```diff
@@ -335,16 +335,16 @@ type ContainerProps = Omit<
...
 
 const ContextPickerModalContainer = createReactClass<ContainerProps, ContainerState>({
   displayName: 'ContextPickerModalContainer',
   mixins: [Reflux.connect(OrganizationsStore, 'organizations') as any],
   getInitialState() {
+    const storeState = OrganizationStore.get();
     return {
-      selectedOrganization:
-        OrganizationStore.organization && OrganizationStore.organization.slug,
+      selectedOrganization: storeState.organization && storeState.organization.slug,
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx:347]:
```suggestion
      selectedOrganization: storeState.organization?.slug,
```

**Review by matejminar — APPROVED** [review]

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 2]
```diff
@@ -335,16 +335,16 @@ type ContainerProps = Omit<
 
 type ContainerState = {
   organizations?: Organization[];
-  selectedOrganization?: string;
+  selectedOrganization: string | null;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx:338]: should we move this type to the top as it is not optional?

**Review by priscilawebdev — APPROVED** [review]

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/contextPickerModal.tsx]: I applied this but as a separate commit as I also had to adjust the State type.

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 3]
```diff
@@ -334,17 +334,17 @@ type ContainerProps = Omit<
 };
 
 type ContainerState = {
+  selectedOrganization: string | undefined;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx:337]:
```suggestion
  selectedOrganization?: string;
```

**Pull request merged**